### PR TITLE
fixing getVersion() to handle null case of versionId

### LINKED ISSUE #2903: Remove ComponentContext.documentId, Container.id
`documentId` is a Container level concept and should not be exposed via ComponentContext and ComponentRuntime.
Especially with detachedContainers, Components do not know (or should care) when the Container is attached and the documentId is not available before the Container is attached.

### PR DESCRIPTION
**Pull request by chensixx**

### CODE REVIEW

File: packages/drivers/file-driver/src/fileDocumentStorageService.ts [hunk 1]
```diff
@@ -76,8 +76,11 @@ export class FluidFetchReader extends ReadDocumentStorageServiceBase implements
      * @param versionId - version ID.
      * @param count - Number of versions to be returned.
      */
-    public async getVersions(versionId: string, count: number): Promise<api.IVersion[]> {
-        if (versionId === FileStorageDocumentName) {
+    public async getVersions(versionId: string | null, count: number): Promise<api.IVersion[]> {
+        if (versionId === null){
```
Comment (vladsud) [inline packages/drivers/file-driver/src/fileDocumentStorageService.ts:80]: I'd rather roll it into next IF. There is some logic there that may return [] that should apply as equally to null option as it does for FileStorageVersionTreeId

**Review by vladsud — APPROVED** [review]
Love the change!
It would be great to open follow up issue to remove handling of docId from drivers. But that can be done only after some delay, i.e. loader changes that start using null are consumed and propagated to all clients.

**Comment by chensixx** [comment]
i changed all getVersions() to take string | null

File: packages/drivers/file-driver/src/fileDocumentStorageService.ts [hunk 2]
```diff
@@ -143,18 +143,19 @@ export const FileSnapshotWriterClassFactory = <TBase extends ReaderConstructor>(
...
         }
 
-        public async getVersions(versionId: string, count: number): Promise<api.IVersion[]> {
+        public async getVersions(versionId: string | null, count: number): Promise<api.IVersion[]> {
             // If we already saved document, that means we are getting here because of snapshot generation.
             // Not returning tree ensures that ContainerRuntime.snapshot() would regenerate subtrees for
             // each unchanged data store.
             // If we want to change that, we would need to capture docId on first call and return this.latestWriterTree
             // when latest is requested.
+            if (this.latestWriterTree && (this.docId === versionId || versionId === undefined) || versionId === null) {
```
Comment (vladsud) [inline packages/drivers/file-driver/src/fileDocumentStorageService.ts:152]: I think this needs to be 

> if (this.latestWriterTree && (this.docId === versionId || versionId === null)) {

I.e. null should be handled same was as docId. And I believe we dropped undefined (which mean no snapshot) long time ago (worth double checking, but clearly API shape does not allow that)

File: packages/drivers/file-driver/src/fileDocumentStorageService.ts [hunk 3]
```diff
@@ -143,18 +143,19 @@ export const FileSnapshotWriterClassFactory = <TBase extends ReaderConstructor>(
...
+                return [{ id: "latest", treeId: FileStorageVersionTreeId }];
+            }
+
             if (this.docId === undefined && versionId !== undefined) {
```
Comment (vladsud) [inline packages/drivers/file-driver/src/fileDocumentStorageService.ts:156]: This condition needs to be changed as well to exclude null (you can change undefined -> null given that we no longer support undefined)

**Comment by vladsud** [comment]
Any chance you can glance at why compiler was not complaining about getVersions() having different shape from interface? It feels like some warnings are disabled, and maybe it's worth taking a look (or opening a bug) to follow up here.

**Review by vladsud — APPROVED** [review]

**Pull request merged**